clients/itwin/src/Client.ts

```diff
@@ -71,175 +69,44 @@ export abstract class Client {
     return Client._defaultRequestOptionsProvider.assignOptions(options);
   }
 
-  /**
-   * Implemented by clients to specify the name/key to query the service URLs from
-   * the URL Discovery Service ("Buddi")
-   * @returns Search key for the URL.
-   */
-  protected abstract getUrlSearchKey(): string; // same as the URL Discovery Service ("Buddi") name
-
   /**
    * Gets the URL of the service. Attempts to discover and cache the URL from the URL Discovery Service. If not
    * found uses the default URL provided by client implementations. Note that for consistency
    * sake, the URL is stripped of any trailing "/"
    * @returns URL for the service
    */
-  public async getUrl(requestContext: ClientRequestContext): Promise<string> {
+  public async getUrl(): Promise<string> {
     if (this._url)
       return this._url;
 
-    if (this.baseUrl) {
-      let prefix = process.env.IMJS_URL_PREFIX;
-
-      // Need to ensure the usage of the previous IMJS_BUDDI_RESOLVE_URL_USING_REGION to not break any
-      // existing users relying on the behavior.
-      // This needs to be removed...
-      if (undefined === prefix) {
-        const region = process.env.IMJS_BUDDI_RESOLVE_URL_USING_REGION;
-        switch (region) {
-          case "102":
-            prefix = "qa-";
-            break;
-          case "103":
-            prefix = "dev-";
-            break;
-        }
-      }
-
-      if (prefix) {
-        const baseUrl = new URL(this.baseUrl);
-        baseUrl.hostname = prefix + baseUrl.hostname;
-        this._url = baseUrl.href;
-      } else {
-        this._url = this.baseUrl;
-      }
-
-      // Strip trailing '/'
-      this._url = this._url.replace(/\/$/, "");
-      return this._url;
+    if (!this.baseUrl) {
+      throw new Error("need base url");
     }
 
-    const urlDiscoveryClient: UrlDiscoveryClient = new UrlDiscoveryClient();
-    const searchKey: string = this.getUrlSearchKey();
-    try {
-      const url = await urlDiscoveryClient.discoverUrl(requestContext, searchKey, undefined);
-      this._url = url;
-    } catch (error) {
-      throw new Error(`Failed to discover URL for service identified by "${searchKey}"`);
+    let prefix = process.env.IMJS_URL_PREFIX;

```

Comment:
We should document a list of all env vars used in core. One of the pains with the buddi env vars was no one knew what they were supposed to be unless you were familiar with the code

Reply:
Yes, agreed. I'm debating if we remove all env vars that are magic and instead have each client take in an optional url for the service endpoint. That way when an app is initializing the client it can configure what endpoint it wants to use and it's not magic...

I think in a subsequent PR I may remove this entire `Client` class.